department-of-veterans-affairs/caseflow
# Piedram/appeals 21083

Resolves [APPEALS-21082](https://vajira.max.gov/browse/APPEALS-21082), [APPEALS-21606](https://vajira.max.gov/browse/APPEALS-21606), [APPEALS-19797](https://vajira.max.gov/browse/APPEALS-19797), [APPEALS-21083](https://jira.devops.va.gov/browse/APPEALS-21083)

### Description

As a developer, I need to add backend functionality to the “Cancel Tasks and Reassign” button on the Confirm Cancellation modal so that the user can select their desired functionality.   

### Acceptance Criteria:

- [ ] New Assign task is created and active 

All the selected data from the confirm cancellation Modal should be saved in a DB table (VACOLS + Caseflow)
- [ ]  priorloc and brieff VACOLS tables will be updated with re-assignment.
- [ ]  Create a new task with reasons, new judge, and instructions + context.
- [ ] All blocking tasks will be cancelled.
- [ ] New Task is found in the "Currently active tasks" section.
- Assigned To: <VLJ>
- Assigned By: <Board user>
- Task.
- Task Instructions.
- Reason.
- New Judge.
- Details.

### Testing Plan

1. Go to Caseflow login and sign in as Special Case Movement user,  for example, `BVARDUNKLE (VACO)`
2. Navigate `https://voyager.caseflowdemo.com/queue/appeals/#`  any legacy appeal under  `Distribution Task`
3. In the `Currently active tasks` click in the actions dropdown and you have to see the option  `Case Movement: Advance to judge`.

![11](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/0dd685fd-cc0f-4859-ab94-6c27a70caebf)

4. Fill up the information that is required on this page.

![12](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/368d19d7-2b01-4600-9884-31f9110ad590)

![13](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/d115ee9d-b32b-4a5f-8c9f-9c764e5cc275)

5.  Click on `Cancel Task and Reassign` Button

![14](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/548a4626-2fd9-4f6a-8ee0-f3a68f985902)

6. Now you have to see the Hearing tasks were canceled and the task is already assigned.

7. The task tree looks like this.

![16](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/307d916d-ac4b-4671-a474-fa5266137fb2)


### Testing Plan for Travel type 

1. If the Hearing type is travel all the functionality is the same, But when we cancel and advance to judge, the application create again a hearing type and is put on `hold` This is required for COVID-19. and it is ok.
2.  when the process finish you have to see like this

![especial11](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/d3620db3-3253-4bfa-9a75-b00926c35c65)

4. the Task tree looks like these


![especial12](https://github.com/department-of-veterans-affairs/caseflow/assets/110848569/cd7606a4-111e-4bf2-90d8-eb9c6137f6cc)
